Sources/PortalSwift/Core/Mpc/Mobile/EnclaveMobileWrapper.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  EnclaveMobileWrapper.swift
+//  PortalSwift
+//
+//  Created by Rami Shahatit on 2/19/25.
+//
+
+import Foundation
+
+class EnclaveMobileWrapper: MPCMobile {
+  private let requests: PortalRequestsProtocol
+
+  init(
+    requests: PortalRequestsProtocol = PortalRequests()
+  ) {
+    self.requests = requests
+  }
+
+  // Override sign method to use HTTP endpoint
+  func MobileSign(
+    _ apiKey: String?,
+    _: String?,
+    _ signingShare: String?,
+    _ method: String?,
+    _ params: String?,
+    _ rpcURL: String?,
+    _ chainId: String?,
+    _ metadata: String?
+  ) async -> String {
+    guard let apiKey = apiKey,
+          let signingShare = signingShare,
+          let method = method,
+          let params = params,
+          let rpcURL = rpcURL,
+          let chainId = chainId,
+          let metadata = metadata
+    else {
+      return encodeErrorResult(id: "INVALID_PARAMETERS", message: "Invalid parameters provided")
+    }
+
+    guard let url = URL(string: "https://mpc-client.portalhq.io/v1/sign") else {
+      return encodeErrorResult(id: "INVALID_URL", message: "Invalid URL")
+    }
+
+    let requestBody: [String: String] = [
+      "method": method,
+      "params": params,
+      "share": signingShare,
+      "chainId": chainId,
+      "rpcUrl": rpcURL,
+      "metadataStr": metadata,
+      "clientPlatform": "NATIVE_IOS",
+      "clientPlatformVersion": SDK_VERSION
+    ]
+
+    do {
+      let data = try await requests.post(url, withBearerToken: apiKey, andPayload: requestBody)
+      let enclaveResponse = try JSONDecoder().decode(EnclaveSignResponse.self, from: data)
+      return encodeSuccessResult(data: enclaveResponse.data)
+
+    } catch {
+      return encodeErrorResult(id: "SIGNING_NETWORK_ERROR", message: error.localizedDescription)
```

Comment:
catch the potential 400 response from the api and then pass the id and message from this error